### ISSUE TITLE
Line plot with constant value results in infinite loop

### ISSUE THREAD

**Issue by konn** [cchalmers/plots#51]
Hi, thank you for great package!

It seems that if one feeds constant input to `linePlot` function, it produces NaN in the final diagram and makes `diagrams` hang forever when it is written out as a raster data.

We are using plots library to create a debug information for some simulation and I stumbled upon this bug in some specific settings which results in constant inputs.

Reproduction:

```hs
import qualified Data.Vector.Unboxed as U
import Diagrams.Backend.Rasterific qualified as Raster
import Diagrams.Prelude
import Diagrams.TwoD.Text qualified as Dia

main :: IO ()
main = 
  Raster.renderRasterific ("output.png") (mkWidth 512) $ lineAlongY "constant" [[U.fromList [0, 0, 0, 0]]

lineAlongY ::
  (Renderable (Path V2 Double) b, Renderable (Dia.Text Double) b) =>
  String ->
  [[U.Vector Double]] ->
  QDiagram b V2 Double Any
lineAlongY name =
  Compose
    >>> ifoldMap
      ( \(uncurry (+) -> !i) ->
          U.indexed >>> U.foldMap \(j, x) ->
            DL.singleton $ P $ V2 (fromIntegral $ i + j) x
      )
    >>> ( \pts ->
            r2Axis &~ do
              linePlot (DL.toList pts) $ plotColor .= rgb 0 0.25 1.0
              titleText .= name
        )
    >>> renderAxis
    >>> extrudeLeft 100
    >>> bg white
```

**Comment by cchalmers** [comment]
Thanks for the report! I've pushed a fix and upload 0.1.1.5: https://hackage.haskell.org/package/plots-0.1.1.5.

It was inferring empty bounds in the y-axis. I've made it so it's at least 1e-6 each way.

You code was missing some bits for me compile it:

```haskell
{-# LANGUAGE BlockArguments #-}
{-# LANGUAGE FlexibleContexts #-}
{-# LANGUAGE OverloadedStrings #-}
{-# LANGUAGE TypeApplications #-}
{-# LANGUAGE TypeFamilies #-}
{-# LANGUAGE TypeOperators #-}
{-# LANGUAGE ViewPatterns #-}

import qualified Data.Vector.Unboxed as U
import Diagrams.Backend.Rasterific qualified as Raster
import Diagrams.Prelude
import Diagrams.TwoD.Text qualified as Dia
import Plots
import Data.Colour.SRGB.Linear

import Control.Arrow
import Data.Functor.Compose

main :: IO ()
main =
  Raster.renderRasterific ("output.png") (mkWidth 512) $ lineAlongY "constant" [[U.fromList [0, 0, 0, 0]]]

lineAlongY ::
  (Renderable (Path V2 Double) b, Renderable (Dia.Text Double) b) =>
  String ->
  [[U.Vector Double]] ->
  QDiagram b V2 Double Any
lineAlongY name =
  Compose
    >>> ifoldMap
      ( \(uncurry (+) -> !i) ->
          U.indexed >>> U.foldMap \(j, x) ->
            (:[]) $ P $ V2 (fromIntegral $ i + j) x
      )
    >>> ( \pts ->
            r2Axis &~ do
              linePlot pts $ plotColor .= rgb 0 0.25 1.0
              titleText .= name
        )
    >>> renderAxis
    >>> extrudeLeft 100
    >>> bg white
```

![output](https://github.com/user-attachments/assets/7e41eb8d-9dfd-49b0-91a9-8c76419c94a8)

**Comment by konn** [comment]
Thank you for quick reply!
